nativelib/src/main/scala/java/lang/resource/EmbeddedResourceHelper.scala

```diff
@@ -0,0 +1,29 @@
+package java.lang.resource
+
+import java.util.Base64
+
+private[lang] object EmbeddedResourceHelper {
+
+  lazy val resourceFileIdMap =
+    EmbeddedResourceHelper.getAllFilePaths().zipWithIndex.toMap
+
+  // Decodes, constructs and returns all embedded resource file paths.
+  private def getAllFilePaths(): Array[String] = {
+    val filePathAmount = EmbeddedResourceReader.getEmbeddedSize()
+    val res = Array.ofDim[Array[Byte]](filePathAmount)
+
+    for (id <- 0 until filePathAmount) {
```

Comment:
Do you want to use a generator vs a simple while loop?

Reply:
Yes, I think you should be able to use `while` loops with counters or `java.util.ScalaOps` to avoid using Scala collections but feel like you are using them. I brought `ScalaOps` over from Scala.js recently and it is pretty much genius.